### PR TITLE
SL-5: Add autocomplete dropdown for items

### PR DESCRIPTION
**Pull request by rupertbarton**

### CODE REVIEW

File: shopping-list-web/src/InputText/InputTest.test.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { fireEvent, render, screen, waitFor } from "@testing-library/react";
 import InputText from "./InputText";
 
+const mockCurrentProduct = ""
```
Comment (sophie-richardson25) [inline shopping-list-web/src/InputText/InputTest.test.tsx]: 🤔 can you not mock it with jest like the others are? Impressed you touched the tests though tbh

File: shopping-list-web/src/InputText/InputText.tsx [hunk 2]
```diff
@@ -1,30 +1,70 @@
...
+          onChange={handleProductInput}
+        />
+        <input
+          type="text"
+          aria-label="amountInput"
+          placeholder="amount"
+          value={props.currentAmount}
+          onChange={(e) => props.setCurrentAmount(e.target.value)}
+        />
+        {!!Object.keys(items).length &&
```
Comment (sophie-richardson25) [inline shopping-list-web/src/InputText/InputText.tsx]: I feel like maybe the dropdown could be a separate component? not sure

File: shopping-list-web/src/InputText/InputText.tsx [hunk 3]
```diff
@@ -1,30 +1,70 @@
...
+          <div className="dropdown-content">
+            {Object.keys(items).map((i) => (
```
Comment (sophie-richardson25) [inline shopping-list-web/src/InputText/InputText.tsx]: could `i` be renamed to something more DESCRIPTIVE pls

File: shopping-list-web/src/ShoppingList/ShoppingList.tsx [hunk 4]
```diff
@@ -40,14 +40,18 @@ export default function ShoppingList() {
         category,
         currentProduct,
         currentAmount
-      ); //TODO: get rid of the product and amount after this
+      );
+      setCurrentProduct("")
```
Comment (sophie-richardson25) [inline shopping-list-web/src/ShoppingList/ShoppingList.tsx]: 😍

File: shopping-list-web/src/utilities/httpMethods/productMethods.ts [hunk 5]
```diff
@@ -20,3 +20,10 @@ export const getCategoryDataResponseForProduct = async (product: string) => {
   const result = await response.json();
   return result;
 };
+
+export const getDataResponseForFilteredProducts = async (query: string): Promise<any> => {
```
Comment (sophie-richardson25) [inline shopping-list-web/src/utilities/httpMethods/productMethods.ts]:
```suggestion
export const getDataResponseForFilteredProducts = async (query: string): Promise<SetStateAction<Map<string, IListItem[]>>> => {
```

File: shopping-list-web/src/InputText/InputText.tsx [hunk 6]
```diff
@@ -1,30 +1,70 @@
...
+    props.setCurrentProduct(current_value)
+    
+  }
+
+  const handleAutoCompleteClick = (item: string) => {
+    props.setCurrentProduct(item)
+  }
+
+  useEffect(() => {
+    const timeOutId = setTimeout(() => getDataResponseForFilteredProducts(props.currentProduct).then((res) => setItems(res)), 500);
```
Comment (sophie-richardson25) [inline shopping-list-web/src/InputText/InputText.tsx]: I don't really know what's going on here but it's very functional and looks cool

File: shopping-list-web/src/InputText/InputText.tsx
Comment (sophie-richardson25) [inline shopping-list-web/src/InputText/InputText.tsx:1]: can we run Prettier over this file please

**Review by sophie-richardson25 — CHANGES_REQUESTED** [review]

File: shopping-list-web/src/InputText/InputText.tsx (hunk 2 above)
Comment (rupertbarton) [reply shopping-list-web/src/InputText/InputText.tsx]: I've moved it out to a separate component!

File: shopping-list-web/src/InputText/InputText.tsx (hunk 3 above)
Comment (rupertbarton) [reply shopping-list-web/src/InputText/InputText.tsx]: Renamed!

File: shopping-list-web/src/utilities/httpMethods/productMethods.ts (hunk 5 above)
Comment (rupertbarton) [reply shopping-list-web/src/utilities/httpMethods/productMethods.ts]: Fixed the types here I think... but not sure how happy you'll be with the interface I made...

File: shopping-list-web/src/InputText/InputText.tsx (hunk 6 above)
Comment (rupertbarton) [reply shopping-list-web/src/InputText/InputText.tsx]: This means that it only sends a request off after 500 milliseconds after the user has finished typing

File: shopping-list-web/src/InputText/InputText.tsx
Comment (rupertbarton) [reply shopping-list-web/src/InputText/InputText.tsx:1]: I've run prettier over everything

File: shopping-list-web/src/common/autocompleteDropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import "./autocompleteDropdown.css";
+
+const AutocompleteDropdown = (props: {
```
Comment (sophie-richardson25) [inline shopping-list-web/src/common/autocompleteDropdown.tsx]: think you can do `export default function AutocompleteDropdown(` and then you don't need to export it at the bottom

File: shopping-list-web/src/utilities/httpMethods/productMethods.ts (hunk 5 above)
Comment (sophie-richardson25) [reply shopping-list-web/src/utilities/httpMethods/productMethods.ts]: I'm not that happy with it but I will allow it for now  😂

**Review by sophie-richardson25 — APPROVED** [review]

**Pull request merged**